<?xml version="1.0"?>

<launch>
  <!-- Crazyflie args -->
  <arg name="cf_name" default="crazyflie" />
  <arg name="uri" default="radio://0/105/2M/0xE7E7E7E701" />
  <arg name="frame" default="crazyflie/crazyflie" />
  <arg name="to_sim" default="False"/>
  <arg name="starting_position" default="[0.0, 0.0, 0.0]" />

  <arg name="take_off_height" default="0.5" />
  <arg name="gnd_height" default="0.0" />

  <!-- Create crazyflie -->
  <group ns="$(arg cf_name)">

    <!-- Add CF to server -->
    <group unless="$(arg to_sim)">

      <include file="$(find crazyflie_driver)/launch/crazyflie_add.launch">
        <arg name="uri" value="$(arg uri)" />
        <arg name="tf_prefix" value="$(arg cf_name)" />
        <arg name="roll_trim" default="0" />
        <arg name="pitch_trim" default="0" />

        <arg name="enable_logging" value="True" />
        <arg name="use_ros_time" value="True" />
        <arg name="enable_logging_imu" value="True" />
        <arg name="enable_logging_temperature" value="True" />
        <arg name="enable_logging_magnetic_field" value="True" />
        <arg name="enable_logging_pressure" value="True" />
        <arg name="enable_logging_battery" value="True" />
        <arg name="enable_logging_pose" value="True" />
        <arg name="enable_logging_packets" value="True" />
      </include>
    </group>

    <!-- CF controller -->
    <node name="$(arg cf_name)_controller" pkg="swarm_manager" type="crazyflie.py" output="screen">
        <param name="cf_name" value="$(arg cf_name)"/>
        <param name="to_sim" value="$(arg to_sim)"/>

        <param name="take_off_height" value="$(arg take_off_height)"/>
        <param name="gnd_height" value="$(arg gnd_height)"/>
    </node>

    <!-- Simulation of CF -->
    <group if="$(arg to_sim)">
      <node name="$(arg cf_name)_sim" pkg="swarm_manager" type="crazyflie_sim.py" output="screen">
        <param name="cf_name" value="$(arg cf_name)"/>
        <rosparam param="starting_position" subst_value="True">$(arg starting_position) </rosparam>
      </node>
    </group>

    <!-- Position frame -->
    <!-- <node pkg="tf" type="static_transform_publisher" name="baselink_broadcaster" args="0 0 0 0 0 0 1 $(arg frame) /$(arg cf_name)/base_link 100" /> -->

    <node name="crazyflie_broadcaster" pkg="swarm_manager" type="crazyflie_tf.py" output="screen">
      <param name="cf_name" value="$(arg cf_name)" />
      <param name="frame" value="$(arg frame)" />
    </node>

  </group>

</launch>